### PR TITLE
Error tweaks for the interpreter & debugger

### PR DESCRIPTION
**Pull request by EclecticGriffin**
A small PR which sets some of the groundwork for #622. Move and expand the `InterpreterError` struct and make it the return type for the interpreter's main.

The debugger exit command now properly exits without an error code.

Updated the error for conflicting assignments to use an actual error rather than a panic. This mostly just means we'll stop having to update the `.expect` files whenever the code line producing the error changes.

### CODE REVIEW

File: interp/runt.toml [hunk 1]
```diff
@@ -84,7 +84,7 @@ cmd = """
 """
 
 [[tests]]
-name = "primitives (CIDB)"
+name = "primitives (CIDR)"
```
Comment (rachitnigam) [inline interp/runt.toml:87]: Name hype!

File: interp/tests/errors/multiple_drivers_comb.expect [hunk 2]
```diff
@@ -1,9 +1,7 @@
 ---CODE---
-101
+1
 ---STDERR---
-thread 'main' panicked at '[interpret_group]: multiple assignments to one port: add0.left
-Between assignments:
-add0.left = const0.out;
-and
-add0.left = const1.out;', interp/src/interpreter/steppers/group_interpreter.rs:178:25
-note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Comment (rachitnigam) [inline interp/tests/errors/multiple_drivers_comb.expect:9]: I like this change!

**Review by rachitnigam — APPROVED** [review]
Cool, straightforward quality of life improvement!  LGTM!

File: interp/src/errors.rs [hunk 3]
```diff
@@ -0,0 +1,79 @@
+use crate::utils::assignment_to_string;
+use calyx::errors::Error;
+use calyx::ir::{Assignment, Id};
+use rustyline::error::ReadlineError;
+use thiserror::Error;
+
+// Utility type
+pub type InterpreterResult<T> = Result<T, InterpreterError>;
+
+#[derive(Error)]
```
Comment (sampsyo) [inline interp/src/errors.rs:10]: Neato; `thiserror` seems like a really useful little crate.

**Review by sampsyo — APPROVED** [review]
Yay!! Nice work—I agree with @rachitnigam that this is an unmitigated win. Wahoo!!

**Pull request merged**